src/3rd_party_add.c

```diff
@@ -93,13 +96,44 @@ enum swupd_code third_party_add_main(int argc, char **argv)
 
 	/* The last two in reverse are the repo-name, repo-url */
 	if (third_party_add_repo(name, url) == 0) {
-		info("Repository %s added successfully\n", argv[argc - 2]);
-	} else {
+		info("Repository %s added successfully\n", name);
+	}
+
+	/* at this point the repo has been added to the repo.ini file */
+	repos = third_party_get_repos();
+	repo = list_search(repos, name, repo_name_cmp);
+	if (!repo) {
 		ret = SWUPD_COULDNT_WRITE_FILE;
-		error("Failed to add repo: %s to config\n", argv[argc - 2]);
+		error("Failed to add repo: %s to config\n", name);
+		goto finish;
+	}
+
+	/* set the appropriate content_dir and state_dir for the selected 3rd-party repo */
+	ret = third_party_set_repo(globals.state_dir, globals.path_prefix, repo);
+	if (ret) {
+		goto finish;
 	}
 
+	/* get repo's latest version */
+	repo_version = get_latest_version(repo->url);
+	if (repo_version < 0) {
+		error("Unable to determine the latest version for repository %s\n\n", repo->name);
+		ret = SWUPD_INVALID_REPOSITORY;
+		goto finish;
+	}
+
+	/* the repo's "os-core" bundle needs to be installed at this moment
+	 * so we can track the version of the repo */
```

Comment:
I'm sadly a little late to this. Is the os-core bundle from a 3rd-party repository essentially the /usr/lib/os-release file?

Reply:
We have been going back and forth on adding an os-core with only the os-release file or tracking the version number ourselves. Last conclusion was that it would be better to have an os-core with only os-release file, but I'm open to discuss that.
It's not late.

Reply:
Seems reasonable to me, thanks =).